source/server/SocketClient.cpp

```diff
@@ -104,6 +104,21 @@ nn::Result SocketClient::init(const char* ip, u16 port) {
     // on a reconnect, resend some maybe missing packets
     if (initPacket.conType == ConnectionTypes::RECONNECT) {
       client->resendInitPackets();
+    } else {
+        // empty TagInf
+        TagInf tagInf;
```

Comment:
Why are we sending a new tag packet on first connect. We dont send any new costume packets right?

Reply:
The costume packets are send after **every** stage load, so not really needed here in my opinion.

Example:
- Someone's game crashes.
- They start the game again.
- Connect again to the same server (init packets).
- [New] Send empty Tag and Capture packets to reset their state cached by other players to the default when starting a new game. (Otherwise the game of the other players might still think they are inside of a capture or in H&S seeker mode.)
- They continue their save or create a new game.
- Loading time.
- [As Before] Send game packet (stage & scenario).
- [As Before] Send costume packet.

Reply:
Also if we would send the costume packet directly after the connection we can only send the default Mario costume packet, because we're still in the main menu and don't know which save and therefore which costume they will be using.

But sending that default Mario costume will lead to sync issues with other players. Because most likely they will reload the save from their last connection, and therefore the same costume as before, so keeping the old costume in the cache of the other players is most likely correct (though not alwasy). But when we send the default mario costume and in the time between the connection and sending the correct costume packet after the stage load, another player changes their stage, their puppet socket will load the default mario costume for the player instead of the old costume, and will need to reload their stage to fix the costume for the re-connecting player again.

Reply:
Yeh, that makes sense to me then.